New Police.uk email domain

LeePorte opened the pull request:
What
----

There is a plan to move all police email address to fall under one domain (police.uk). This commit ensures that these domains are valid for submitting whilst ensuring that we reject non police domains.

How to review
-------------

Code review and check the tests pass

Who can review
--------------

Anyone, but possibly @AP-Hunt due to prior context

Commit:
New Police.uk email domain

There is a plan to move all police email address to fall under one domain (police.uk). This commit ensures that these domains are valid for submitting whilst ensuring that we reject non police domains.

AP-Hunt commented on spec/signup_spec.rb:
```diff
@@ -132,6 +132,17 @@
 		expect(page.status_code).to eq(400)
 	end
 
+  it "should require a @location.police.uk or @police.uk email for person_email" do
```
`location.police.uk` is now covered by the more general `police.uk` domain, I think?  No need to specify in that case

Commit:
Update test name to more accurately describe reality

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

Commit:
Add failing test for partial tlds

This test shows us a bug

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

Commit:
Make the domain regular expression less confusing

I am not particularly hot with regular expressions

I prefer to use ruby's Regexp.union method to combine regular
expressions, instead of having one big one, I find it easier to
understand this way

In the previous commit I wrote a test which exposes a bug, which this
commit now fixes

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>

tlwr commented:
There was a bug which I have written a test for and now hopefully fixed

AP-Hunt commented on spec/signup_spec.rb:
```diff
@@ -132,16 +132,18 @@
-		expect(page.first('.form-group--person_email .error-message').text).not_to be_empty
-		expect(page.status_code).to eq(400)
-	end
+  %w(gov.uk nhs.net nhs.uk mod.uk police.uk).each do |tld|
```
I think we could add a second test here for accepting subdomains of our allowed list of domains, since it'd be easy to cause a regression in that when editing the regular expressions

tlwr replied:
done in 9b575bf

Commit:
add more tests to cover subdomains

Signed-off-by: Toby Lorne <toby.lornewelch-richards@digital.cabinet-office.gov.uk>
